### PR TITLE
Change sync_blocks to pick smarter maxheight (on top of #9196)

### PR DESCRIPTION
**Pull request by ryanofsky**
This PR changes the `sync_blocks` test function to make it more reliable.

The change is based on #9196 to prevent a test failure that would otherwise start to happen.

### CODE REVIEW

**Comment by ryanofsky** [comment]
Rearranged if-conditions as suggested offline by @morcos.

File: qa/rpc-tests/test_framework/util.py [hunk 1]
```diff
@@ -129,17 +129,19 @@ def sync_blocks(rpc_connections, *, wait=1, timeout=60):
     one node already synced to the latest, stable tip, otherwise there's a
     chance it might return before all nodes are stably synced.
```
Comment (MarcoFalke) [inline qa/rpc-tests/test_framework/util.py]: Does this comment still apply, considering that you switched from `0` to `getblockcount`?

**Comment by MarcoFalke** [comment]
Concept ACK

File: qa/rpc-tests/test_framework/util.py (hunk 1 above)
Comment (ryanofsky) [inline qa/rpc-tests/test_framework/util.py]: This change might make the problem less likely to happen, but it still could happen. If none of the nodes in the set are fully synced when the `sync_blocks` call is made, then `maxheight` will be set to some height less than the final stable height, and the call return could before the nodes are all synced.

**Comment by morcos** [comment]
ACK

**Comment by MarcoFalke** [comment]
@ryanofsky The 60cf51c patch looks like a separate bugfix, not strictly related to tests. Would you mind submitting this as another pull request?

**Pull request merged**